src/config.ts

```diff
@@ -32,6 +32,7 @@ export type ConditionConfig = {
   blockingLabels: string[],
   requiredBodyRegex: string | undefined
   blockingTitleRegex: string | undefined
+  requiredAuthorRole: string
```

Comment:
```suggestion
  requiredAuthorRole: CommentAuthorAssociation
```

Reply:
There should be a decoder that converts/validates string to CommentAuthorAssociation:

```js
const commentAuthorAssociationDecoder: Decoder<CommentAuthorAssociation> = oneOf(constant<'MEMBER'>('MEMBER'), constant<'OWNER'>('OWNER'), constant<'COLLABORATOR'>('COLLABORATOR'), ...);
```

Reply:
I tried this but I get
```
    src/config.ts:79:7 - error TS2322: Type 'Decoder<"MEMBER" | "OWNER" | "NONE">' is not assignable to type 'Decoder<CommentAuthorAssociation>'.
      Type '"MEMBER" | "OWNER" | "NONE"' is not assignable to type 'CommentAuthorAssociation'.
        Type '"MEMBER"' is not assignable to type 'CommentAuthorAssociation'.
```
I don't see where it converts from `string` to `CommentAuthorAssociation`.
I tried `oneof(contant<CAA.NONE>('NONE'), ...)` and it returns
```
    src/config.ts:80:44 - error TS2345: Argument of type '"OWNER"' is not assignable to parameter of type 'CommentAuthorAssociation.OWNER'.

    80   constant<CommentAuthorAssociation.OWNER>('OWNER'),
```
I'm missing something obvious, both sorry, I'm not familiar with the language.

Reply:
I had to dive in the code to refresh again what was up with strings vs enums again. I made a few commits to resolve the issue. Basically consistently using enums everywhere will avoid the type error you saw.